src/domain/challenge/challenge.resolver.mutations.ts

```diff
@@ -162,4 +164,21 @@ export class ChallengeResolverMutations {
       organisationID
     );
   }
+
+  @Roles(
```

Comment:
We need to introduce new role - `Authenticated` which means that the user must be authenticated, but it is not necessary to be member of any group.

Reply:
Or we have this? Using `Guard` without `Roles`?

Reply:
would prefer not to have another group that is equivalent to being a User - by implication is there is a User role, which is anyone that is authenticated. The guard may need to be smart to be able to understand this.

@Valentin: your thoughts?

Reply:
@techsmyth @atstojanov The guard may not be that smart at the moment, but if that's the way we want to go, we can switch the behavior in the user : role matching and add a check for a user with no roles. We can default the behavior to anonymous if the user is authenticated and not in any roles.

Reply:
key is that there is a difference between an non-authenticated + authenticated user. So it may be that there is a "known" term authenticated that the guard understands - or we create another guard that simply checks for authenticated?

Reply:
We can consider adding another role (restricted group) but definitely not another guard for that.

Reply:
not another restricted group please :)

how about another guard isAuthenticated?